partner/WHMCS/modules/servers/jetpack/jetpack.php

```diff
@@ -0,0 +1,365 @@
+    ];
+}
+
+
+/**
+ * Basic configuration options required for a partner to get a Jetpack plan provisioned. Currently a partner
+ * client id and secret are the only host partner options needed to get an access token to provision a Jetpack plan
+ * @return array
+ */
+function jetpack_ConfigOptions()
+{
+    return [
+        'Jetpack Partner Client ID' => [
+            'Type' => 'text',
+            'Size' => '256',
+        ],
+        'Jetpack Partner Client Secret' => [
+            'Type' => 'text',
+            'Size' => '256',
+        ]
+    ];
+}
+
+
+/**
+ * Equivalent to /provision. Create a Jetpack plan using
+ * a Jetpack Hosting partner account. WHMCS expects the string "success"
+ * to be returned if the process is completed successfully otherwise
+ * a string can be returned which will be logged as part of the error.
+ *
+ * Pre Provisioning Steps:
+ *  Module requirements are validated and will return an error string
+ *  if the module was not setup correctly. An error string will also be
+ *  returned in the even that a request to provision a new returns a
+ *  4xx or 5xx error.
+ *
+ *  An Access token will also be retrieved before trying to provisioning.
+ *  Errors strings are prefixed with 'JETPACK MODULE' so if getting the access
+ *  token starts with this return this as an error.
+ *
+ * If the response from provisioning does not contain "success" in the message
+ * consider the provisioning to be incomplete and get a useful error from the response
+ *
+ * @param array WHMCS $params
+ * @return string Either 'success' or an error with what went wrong when provisioning
+ */
+function jetpack_CreateAccount(array $params)
+{
+
+    $module_errors = validate_required_fields($params);
+    if ($module_errors !== true) {
+        return $module_errors;
+    }
+
+    $access_token = get_access_token($params);
+    if (strpos($access_token, 'JETPACK MODULE') === 0) {
+        return $access_token;
+    }
+
+    $provisioning_url = "https://public-api.wordpress.com/rest/v1.3/jpphp/provision";
+    $stripped_url = preg_replace("(^https?://)", "", $params['customfields']['Site URL']);
+    $stripped_url = rtrim($stripped_url, '/');
+
+    $request_data = [
+        'plan' => strtolower($params['customfields']['Plan']),
+        'siteurl' => $stripped_url,
+        'local_user' => $params['customfields']['Local User'],
+        'force_register' => true,
+    ];
+
+    $response = make_api_request($provisioning_url, $access_token, $request_data);
+    if ($response->success && $response->success == true) {
+        if ($response->next_url) {
+            save_provisioning_details($response->next_url, $params);
+        } elseif (!$response->next_url && $response->auth_required) {
+            save_provisioning_details($response->next_url, $params, true);
+        }
+        return 'success';
+    } else {
+        $errors = get_provisioning_errors_from_response($response);
+        return $errors;
+    }
```

Comment:
I like the way this conditional is setup. If I were to write it, I think I'd do:

```
if ( ! empty() ) {
  // stuff
} else {
  // stuff
}
```  

But, it reads well and we know that `success` is getting returned when expected now. 👍

Reply:
I made a small change to this to help make things less confusing and removed an unneeded check for pending accounts.